SignalMessaging/utils/ThreadUtil.m

```diff
@@ -318,13 +306,14 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
         // have been marked as read.
         //
         // IFF this variable is non-null, there are unseen messages in the thread.
-        if (firstUnseenInteractionTimestampParameter) {
-            result.firstUnseenInteractionTimestamp = firstUnseenInteractionTimestampParameter;
+        NSNumber *_Nullable firstUnseenInteractionTimestamp = nil;
+        if (lastUnreadIndicator) {
```

Comment:
This is so we don't "shift" the unread indicator down while the conversation is open, right?

Reply:
Right - it's about continuity.